packages/protocol/test/governance/validators.ts

```diff
@@ -2184,4 +2184,129 @@ contract('Validators', (accounts: string[]) => {
       })
     })
   })
+
+  describe('#groupMembershipInEpoch', async () => {
+    const validator = accounts[0]
+    const groups = accounts.slice(1, -1)
+
+    beforeEach(async () => {
+      await registry.setAddressFor(CeloContractName.DowntimeSlasher, validator)
+      await registry.setAddressFor(CeloContractName.DoubleSigningSlasher, accounts[3])
+      await registry.setAddressFor(CeloContractName.Governance, accounts[4])
+    })
+
+    describe('when the validator is added to different groups sequentially', async () => {
+      const epochs = []
+      beforeEach(async () => {
+        await registerValidator(validator)
+        for (const group of groups) {
+          await registerValidatorGroup(group)
+        }
+        for (let i = 0; i < membershipHistoryLength.plus(1).toNumber(); i++) {
+          const blockNumber = await web3.eth.getBlockNumber()
+          const epochNumber = Math.floor(blockNumber / EPOCH)
+          const blocksUntilNextEpoch = (epochNumber + 1) * EPOCH - blockNumber
+          epochs.push(epochNumber)
+          await mineBlocks(blocksUntilNextEpoch, web3)
+
+          if (i != 4) {
+            await validators.affiliate(groups[i])
+            await validators.addFirstMember(validator, NULL_ADDRESS, NULL_ADDRESS, {
+              from: groups[i],
+            })
+          } else {
+            await validators.forceDeaffiliateIfValidator(validator)
```

Comment:
You don't need to force deaffiliate, you can just call deaffiliate from the validator address